### ISSUE TITLE
Love the system so far

### ISSUE THREAD

**Issue by bluenevus** [sandstorm-io/sandstorm#2541]
I've been evaluating this and I have to say...it works well.  Yeah, I could docker it but this is much easier.

Apps Wish list...
mindmaps - http://drichard.org/mindmaps/
Vector Graphics - https://github.com/SVG-Edit/svgedit
Web Conferencing - https://github.com/loowid/loowid
Presentation software a bit better than hackerslides - https://github.com/tantaman/Strut
Realtime Collaborative drawing - https://github.com/MicroHealthLLC/mDraw

Thanks once again for the awesome work

**Comment by ocdtrekkie** [comment]
It'll probably make sense to move this to the group list for discussion since it isn't an "issue", but some preliminary thoughts:

mindmaps, SVG-Edit, and Strut are both client-only applications, so someone would need to modify them to save to the server instead of your browser's localStorage. mDraw sounds like it'll package for Sandstorm outright since it uses Redis. Don't know enough about WebRTC to comment on loowid.

**Comment by kpreid** [comment]
Perhaps there could be a “generic server-side storage” library that could be added to apps that are designed around localStorage? (Of course it would still require modifying the apps to use it and support asynchronous access, which could be nontrivial.)

**Comment by ocdtrekkie** [comment]
Yeah, if there was some sort of drop-in replacement for localStorage functionality, where I could go into a client-only HTML5/JS app and replace instances of localStorage with that code fairly easily, there's a lot of apps that could really quickly be ported to Sandstorm, I know of a half dozen others off-hand myself.

While sure, modifications would need to be made if you wanted realtime collaboration or something similar, a lot of apps I can think of handle saving reasonably on their own as it is. A lot of nifty webgames have nifty highscore mechanics that just need a one-time save at the end of a game or something too.

**Comment by bluenevus** [comment]
May not be real-time but you can always share the grain which is async collab…still very useful

--
Vr, Frank

**Comment by ocdtrekkie** [comment]
There's an interesting point in that Sandstorm should probably have some way for apps to specify their grains do not support multiple simultaneous users, and locking them while another user has them open.

**Comment by kentonv** [comment]
https://github.com/sandstorm-io/sandstorm-rawapi-example is a good starting point for client-side apps. The localStorage calls need to be replaced with GET/PUT requests back to the app server, which in principle should be easy. This is how the draw.io app was built.

Real-time collaboration is, of course, a challenge.

**Comment by ocdtrekkie** [comment]
The issue is that using the rawapi example requires building Cap'n Proto from source and such. Definitely amongst all the packaging methods available today... the hardest to implement.

**Comment by bluenevus** [comment]
how about this too?  Its like an open source If This Then That

https://github.com/bipio-server/bipio 
with their pods
https://github.com/bipio-server?page=1

They have a docker so it seems like its possible
https://github.com/bipio-server/bipio-docker

**Comment by ocdtrekkie** [comment]
Something like that would definitely be possible in Sandstorm though my notion is that it'd be a lot easier when the Powerbox feature is more mature, so that it does not require arbitrary networking capabilities.

**Comment by bluenevus** [comment]
another lofty wish!!!!!!!

https://github.com/Open365/Open365

a user can dream can't he?
